app/components/button_component.rb

```diff
@@ -1,38 +1,25 @@
 # frozen_string_literal: true
 
-class ButtonComponent < ViewComponent::Base
-  attr_accessor :spacing, :shape, :typography, :color, :animation
```

Comment:
Was this an intentional removal? I put them there so I could do some follow-on clean-up of the UI for different buttons without having to pass in every single class every time.

Reply:
Yep, this was intentional. I don't think having these CSS classes separated is necessary, or even makes a great deal of sense (cue debate about putting the text color class under "color" vs "typography").

I think if we want to have different variants of buttons, we should make reusable variants of the button, like I did in this PR with the Secondary button. We don't need accessors for that.

If and gene we have other use cases that require overriding something more granular, we can work on a solution then.

Reply:
That makes sense, and we can always use custom little factory methods on the componet if a full component would be "too much"